### ISSUE TITLE
Extend single_match to handle return

### ISSUE THREAD

**Issue by mrmonday** [rust-lang/rust-clippy#929]
In the following code:

``` rust
enum Foo {
    Bar(u8), Baz
}

fn main() {
    match Foo::Bar(3) {
        Foo::Bar(_) => { /* do something */ },
        _ => { return; }
    }
}
```

Clippy produces no warning, but it could produce the single_match warning instead.

**Comment by mcarton** [comment]
It was decided in #327 to limit the lint to enum we know will never be extended.
But we could have another allow-by-default lint for that.

**Comment by camsteffen** [comment]
This is now covered by single_match_else
